mailbox/store/src/main/java/org/apache/james/mailbox/store/event/EventFactory.java

```diff
@@ -62,6 +65,10 @@ public MessageMetaData getMetaData(MessageUid uid) {
         public Mailbox getMailbox() {
             return mailbox;
         }
+
+        public Map<MessageUid, MailboxMessage> getCachedMessages() {
+            return cachedMessages;
```

Comment:
so you propagate a mutable structure as a cache ? It looks like a disaster coming soon ...
You should at least define an immutable datastructure and avoid calling it a cache.
A map is not a cache because  you don't have eviction, right ?